File: Programs/Prime Number Checker/PrimeNumber.java
import java.util.Scanner;
import java.lang.Math;

public class PrimeNumber {
	
	public static boolean isPrime(int number) {
		boolean prime=true;
		int i=2;
		while (i<=Math.sqrt(number)){
			if ((number % i)==0 ){
				prime=false;
			}
			i++;
		}
    return prime;
}

	public static void main(String[] args){

		System.out.println ("Prime Number checker");
		System.out.println ("Please enter a number");
		Scanner s = new Scanner(System.in);
		String temp = s.next();
		
		try {
			Integer i = new Integer(temp);
					System.out.println ("number is an integer");
					System.out.println ("The statement 'the number is a prime' is "+isPrime(i));
		} catch (NumberFormatException i){
			System.out.println ("number is not an integer");
		}
		
		
		
	}

}

File: Programs/Prime Number Checker/PrimeNumber.c
// Tests whether a number is prime

#include "stdafx.h"
#include <stdio.h>

#define true 1
#define false 0
typedef int bool;

bool is_prime(int n)
{
	int divisor;

	if (n <= 1)
		return false;
	for (divisor = 2; divisor * divisor <= n; divisor++)
		if (n % divisor == 0)
			return false;
	return true;
}

int main(void)
{
	int n;

	printf("Enter a number: ");
	scanf("%d", &n);
	if (is_prime(n))
		printf("Prime\n");
	else 
		printf("Not prime\n");
	return 0;
}
